snet_cli/commands.py

```diff
@@ -1224,3 +1228,65 @@ def update(self):
                     self._printerr("Creating transaction to add tags {} to service registration...\n".format(
                         [tag.partition(b"\0")[0].decode("utf-8") for tag in add_tags]))
                     cmd.transact()
+
+    def delete(self):
+        network_id = self._get_network()
+
+        if "config" in self.args and self.args.config:
+            service_json_path = self.args.config
+        else:
+            service_json_path = "service.json"
+
+        try:
+            with open(service_json_path) as f:
+                service_json = json.load(f)
+        except Exception as e:
+            self._error("Failed to load {}!".format(service_json_path))
+
+        agent_address = service_json.get('networks', {}).get(network_id, {}).get('agentAddress', None)
+        if agent_address is None:
+            self._error("Service hasn't been deployed to network with id {}".format(network_id))
+
+        self._printout("Getting information about the service...")
+        registry_contract_def = get_contract_def("Registry")
+        registry_address = self._getstring("registry_at")
+
+        (found, _, current_path, current_agent_address, current_tags) = ContractCommand(
```

Comment:
I saw the similar code in previous PR: https://github.com/singnet/snet-cli/pull/41, https://github.com/singnet/snet-cli/blob/5e64057c1a6984dac4191113233e1b2750e6e7a4/snet_cli/commands.py#L1084
I would remove code duplication and create some python method to call this ```getServiceRegistrationByName()```. Actually may be it makes sense to wrap all contract calls by Python methods to not copy/paste these long code snippets. What do you think?

Reply:
Agree!
I did something similar for the `organization` command.
I'll do the same for the `service`.